pkg/registry/core/rest/storage_core.go

```diff
@@ -242,6 +242,9 @@ func (p *legacyProvider) NewRESTStorage(apiResourceConfigSource serverstorage.AP
 			storage[resource+"/eviction"] = podStorage.Eviction
 		}
 		storage[resource+"/ephemeralcontainers"] = podStorage.EphemeralContainers
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			storage[resource+"/resize"] = podStorage.Resize
```

Comment:
Making this a subresource means that existing admission policies that intercept pod create that have opinions about pod resources (or mutating admission that is auto-adjusting resources) will be bypassed (xref https://github.com/kubernetes/kubernetes/issues/113594)

That *may* be ok if:
1. the permission to write to this new subresource is not granted to anyone other than superusers by default (similar to pods/ephemeralcontainers)
2. we document *very* clearly that before a cluster admin grants this permission, they ensure validating / mutating admission policies are updated to also register for and handle `UPDATE pods/resize` requests

Reply:
Good points. TBH I'm still unsure of whether the subresource is the right decision.

Regarding general admission, this is still a problem without a dedicated subresource, right? Webhooks that deal with resources might not be looking at pod updates since they were previously immutable. Most of the motivation for the subresource was that it's easier to restrict access to just resize, but the flip side is also true (easier to prevent resize permission).

+1 for updating quota. Limitrange is racing with https://github.com/kubernetes/kubernetes/pull/124887